src/test/java/net/imglib2/algorithm/math/ImgMathTest.java

```diff
@@ -0,0 +1,727 @@
+		
+		}
+		
+		// Reset
+		minLM = Long.MAX_VALUE;
+		maxLM = 0;
+		meanLM = 0;
+						
+		for ( int i=0; i < n_iterations; ++i ) {
+			final long t0 = System.nanoTime();
+
+			// Compute saturation
+			final Cursor< UnsignedByteType > cr = red.cursor(),
+					                         cg = green.cursor(),
+					                         cb = blue.cursor();
+			
+			final Cursor< FloatType > cs = saturation.cursor(); 
+			
+			while (cs.hasNext()) {
+				final int r = cr.next().getInteger(),
+						  g = cg.next().getInteger(),
+						  b = cb.next().getInteger();
+				final float max = Math.max( r, Math.max( g, b ) ),
+						    min = Math.min( r, Math.min( g, b ) );
+				cs.next().set( 0.0f == max ? 0.0f : (max - min) / max );
+			}
+
+			final long t1 = System.nanoTime();
+			minLM = Math.min(minLM, t1 - t0);
+			maxLM = Math.max(maxLM, t1 - t0);
+			meanLM += (t1 - t0) / (double)(n_iterations);
+		}
+		
+		assertTrue( "low-level", sumAsInts( saturation, 255.0 ) == saturation_sum );
+
+		if ( print ) System.out.println("Low-level saturation performance: min: " + (minLM / 1000.0) + " ms, max: " + (maxLM / 1000.0) + " ms, mean: " + (meanLM / 1000.0) + " ms");
+		
+		return true;
+	}
+	
+	protected boolean testLT()
+	{
+		final ArrayImg< FloatType, ? > img1 = new ArrayImgFactory< FloatType >( new FloatType() ).create( new long[]{ 10, 10 } );
+		final ArrayImg< FloatType, ? > img2 = new ArrayImgFactory< FloatType >( new FloatType() ).create( new long[]{ 10, 10 } );
+
+		// Fill half the image only
+		for ( final FloatType t : Views.interval( img1, new long[]{ 0, 0 }, new long[] { 9, 4 } ) ) // inclusive
+			t.setReal( 5.0f );
+
+		final ArrayImg< FloatType, ? > target = new ArrayImgFactory< FloatType >(new FloatType() ).create( img1 );
+		
+		compute( LT( img2, img1 ) ).into( target );
+		
+		double sum = 0;
+		for ( final FloatType t : target )
+			sum += t.getRealDouble();
+		
+		System.out.println( "Sum LT 1/0: " + sum );
+		
+		// A "1" for every true comparison, one per pixel for half of all pixels
+		return 50 == sum;
+	}
+	
+	protected boolean testGT()
+	{
+		final ArrayImg< FloatType, ? > img1 = new ArrayImgFactory< FloatType >( new FloatType() ).create( new long[]{ 10, 10 } );
+		final ArrayImg< FloatType, ? > img2 = new ArrayImgFactory< FloatType >( new FloatType() ).create( new long[]{ 10, 10 } );
+
+		// Fill half the image only
+		for ( final FloatType t : Views.interval( img1, new long[]{ 0, 0 }, new long[] { 9, 4 } ) ) // inclusive
+			t.setReal( 5.0f );
+
+		final ArrayImg< FloatType, ? > target = new ArrayImgFactory< FloatType >(new FloatType() ).create( img1 );
+		
+		compute( GT( img1, img2 ) ).into( target );
+		
+		double sum = 0;
+		for ( final FloatType t : target )
+			sum += t.getRealDouble();
+		
+		System.out.println( "Sum GT 1/0: " + sum );
+		
+		// A "1" for every true comparison, one per pixel for half of all pixels
+		return 50 == sum;
+	}
+	
+	
+	@Test
+	public void test1() {
+		assertTrue( testImgMath1() );
```

Comment:
What is the reason for such a convoluted "nested" test structure? Why not simply annotating the more tellingly named test methods with `@Test` and make them `void`, and use `assertEquals` instead of `return x == y` with `assertTrue`?

Reply:
I've followed the same style used by @tpietzsch in other packages. Don't care either way. I did find it advantageous when wanting to warm up the JIT prior to a performance test: this setup enabled me to run the method multiple times prior to the measurement.